### DISCUSSION TITLE
How are you Implementing testing?

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#16601] [Q&A] [answered]
Especially for workflow task activities and in scenarios where you add new properties and stuff?

do you use playwright or just c# unit tests?

**Comment by Piedone** [comment, marked as answer]
See https://www.youtube.com/watch?v=CHdhwD2NHBU.

**Comment by hishamco** [comment]
> do you use playwright or just c# unit tests?

Lombiq has a very good UI ToolBox Testing that you can find [here](https://github.com/Lombiq/UI-Testing-Toolbox) which is based on Selenium

Orchard Core Contrib (OCC) started a project a few months ago that uses Playwright, which you can find [here](https://github.com/OrchardCoreContrib/OrchardCoreContrib.Testing). I'm still working to improve this as much as I can, you can also check my demo [here](https://www.youtube.com/watch?v=8P_6MDSi2Pk)
